docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -0,0 +1,452 @@
+
+        Console.WriteLine("Press any key to exit");
+        while(!Console.KeyAvailable)
+        {
+            // Pretend our store has one transaction each 100ms that each sell 4 hats
+            Thread.Sleep(100);
+            s_hatsSold.Add(4);
+
+            // Pretend we also sold 3 coats. For an ObservableCounter we track the value in our variable and report it
+            // on demand in the callback
+            s_coatsSold += 3;
+
+            // Pretend we have some queue of orders that varies over time. The callback for the "orders-pending" gauge will report
+            // this value on-demand.
+            s_ordersPending = s_rand.Next(0, 20);
+
+            // Last we pretend that we measured how long it took to do the transaction (for example we could time it with Stopwatch)
+            s_orderProcessingTimeMs.Record(s_rand.Next(5, 15));
+        }
+    }
+}
+```
+
+Run the new process and use dotnet-counters as before in a 2nd shell to view the metrics:
+
+```dotnetcli
+> dotnet-counters ps
+      2992 dotnet     C:\Program Files\dotnet\dotnet.exe
+     20508 metric-instr E:\temp\metric-instr\bin\Debug\netcoreapp3.1\metric-instr.exe
+> dotnet-counters monitor -p 20508 HatCo.HatStore
+Press p to pause, r to resume, q to quit.
+    Status: Running
+
+[HatCo.HatStore]
+    coats-sold (Count / 1 sec)                        30
+    hats-sold (Count / 1 sec)                         40
+    order-processing-time
+        Percentile=50                                125
+        Percentile=95                                146
+        Percentile=99                                146
+    orders-pending                                     3
+```
+
+This example uses some randomly generated numbers so your values will vary a bit. We can see that hats-sold (the Counter) and
+coats-sold (the ObservableCounter) both show up as a rate. The ObservableGauge, orders-pending, appears
+as an absolute value. Dotnet-counters renders Histogram instruments as three percentile statistics (50th, 95th, and 99th) but other tools may
+summarize the distribution differently or offer more configuration options.
+
+#### Best practices selecting an instrument type
+
+- For counting things, or any other value that solely increases over time, use Counter or ObservableCounter. Choose between Counter and ObservableCounter depending on which
+  is easier to add to the existing code: either an API call for each increment operation or a callback that will read the current total from a variable the code maintains. In
+  extremely hot code paths where performance is important and using <xref:System.Diagnostics.Metrics.Counter.Add%2A> would create >1 million calls/sec/thread, using
+  ObservableCounter may offer more opportunity for optimization.
+
+- For timing things Histogram is usually prefered. Often it is useful to understand the tail of these distributions (90th, 95th, 99th percentile) rather than averages or
```

Comment:
```suggestion
- For timing things Histogram is usually preferred. Often it is useful to understand the tail of these distributions (90th, 95th, 99th percentile) rather than averages or
```